freqtrade/strategy/interface.py

```diff
@@ -381,6 +384,28 @@ def custom_stake_amount(self, pair: str, current_time: datetime, current_rate: f
         """
         return proposed_stake
 
+    def adjust_trade_position(self, trade: Trade, current_time: datetime,
+                              current_rate: float, current_profit: float, min_stake: float,
```

Comment:
I think we should have a `proposed_stake` amount as well.
that would move some complexity from the strategy to internal bot code (calling `self.dp.wallets.` - which i still don't really see as public interface).

and it seems to be used in all samples I've seen so far, so I'd think the need is there.

Reply:
I'd be OK to add that, assuming we don't also call the custom_stake_amount to overwrite the wallet, because we call adjust_trade_position every x seconds and i think the performance just suffers too much. If we pass the wallet amount in then it should be OK.

Reply:
no, it'd be similar/identical to the parameters we pass to `custom_stake_amount` ... 
calling `custom_stake_amount ` would make no sense - that's what the new callback is supposed to do for subsequent calls.

i don't think it makes a huge difference if that calculation is done as part of the strategy or before - unless you have seen / experienced something that i didn't.

Reply:
OK, so resolved this via removing references to internal wallets and replaced that with using the initial orders stake amount.
`
filled_buys = trade.select_filled_orders('buy')
stake_amount = filled_buys[0].cost
`